sklearn/model_selection/tests/test_search.py

```diff
@@ -2420,3 +2422,26 @@ def test_search_cv_verbose_3(capsys, return_train_score):
     else:
         match = re.findall(r"score=[\d\.]+", captured)
     assert len(match) == 3
+
+
+@pytest.mark.parametrize(
+    "SearchCV, param_search",
+    [
+        (GridSearchCV, "param_grid"),
+        (RandomizedSearchCV, "param_distributions"),
+        (HalvingGridSearchCV, "param_grid"),
+    ],
+)
+def test_search_estimator_param(SearchCV, param_search):
+    # test that SearchCV object doesn't change the object given in the parameter grid
+    X, y = make_classification(random_state=42)
+
+    params = {"clf": [LinearSVC(dual="auto")], "clf__C": [0.01]}
+    orig_C = params["clf"][0].C
+
+    pipe = Pipeline([("trs", MinimalTransformer()), ("clf", None)])
+
+    param_grid_search = {param_search: params}
+    SearchCV(pipe, refit=True, cv=2, scoring="accuracy", **param_grid_search).fit(X, y)
+
+    assert params["clf"][0].C == orig_C
```

Comment:
Could we also add an assert for the `C` value of the `clf` in `pipe`? It should be `0.01` I think and difference from the original. It is kind of a trivial statement, but I had to ponder it for a moment to get straight in my head what is the same instance of the estimator and what isn't. So maybe it would make it clearer for people from the future. My very first reaction while reading it for the first time was "hold on, after fitting with refit=True the param should be the one `SearchCV` found!!", which is true but not what this assert is checking.